### ISSUE TITLE
Retraining a saved model on different dataset

### ISSUE THREAD

**Issue by HarishBaki** [dreamquark-ai/tabnet#539]
I am dealing with a specific case where my observational data is limited, but I have years of atmospheric reanalysis data. So, I trained a TabNet model on the longer available data and then saved it as a pre-trained model. With this, I want to train the same model on limited observational data to pass additional information. I know unsupervised pretraining options exist in TabNet, but I intend to do supervised training both times. 

Would anyone be able to help me with this?

# I did something like this.
tabReg = TabNetRegressor()
tabReg.load_model(f'{OUTPUT_DIR}/TabNet_HOLDOUT_Ens_{str(Ens)}.zip') # load a pretrained model
tabReg.optimizer_params['lr'] = 0.005

tabReg.fit(X_train=X_train, y_train=Y_train_trans,
                    eval_set=[(X_train, Y_train_trans), (X_valid, Y_valid_trans)],
                    eval_name=['train', 'valid'],
                    max_epochs=250, batch_size=256,    #bSize_opt.item(), 
                    eval_metric=['rmse'], patience=10,  #mae, rmse
                    loss_fn = torch.nn.MSELoss(),)
                    
# I did a quick test to see if I can train a model by not initializing the next time as follows. 
First, I initialized tabReg from scratch and trained like the following.

tabReg   = TabNetRegressor(n_d = n_d,
                                n_a = n_a,
                                n_steps = n_steps,
                                n_independent = n_independent,
                                n_shared = n_shared,
                                gamma = gamma,
                                verbose=1,seed=randSeed, )
tabReg.fit(X_train=X_train, y_train=Y_train_trans,
                    eval_set=[(X_train, Y_train_trans), (X_valid, Y_valid_trans)],
                    eval_name=[‘train’, ‘valid’],
                    max_epochs=250, batch_size=256,    #bSize_opt.item(),
                    eval_metric=[‘rmse’], patience=10,  #mae, rmse
                    loss_fn = torch.nn.MSELoss(),)
The loss history is
epoch 0  | loss: 0.22915 | train_rmse: 9.6918  | valid_rmse: 10.11569|  0:00:00s
epoch 1  | loss: 0.02308 | train_rmse: 9.36632 | valid_rmse: 9.4682  |  0:00:01s
epoch 2  | loss: 0.012   | train_rmse: 9.16359 | valid_rmse: 9.36254 |  0:00:02s
epoch 3  | loss: 0.0093  | train_rmse: 7.34215 | valid_rmse: 7.45125 |  0:00:03s
epoch 4  | loss: 0.00837 | train_rmse: 5.16882 | valid_rmse: 5.29309 |  0:00:03s
epoch 5  | loss: 0.00819 | train_rmse: 4.13819 | valid_rmse: 4.18128 |  0:00:04s
epoch 6  | loss: 0.00766 | train_rmse: 2.89976 | valid_rmse: 2.91836 |  0:00:05s
epoch 7  | loss: 0.00742 | train_rmse: 2.02085 | valid_rmse: 2.03557 |  0:00:05s
epoch 8  | loss: 0.00725 | train_rmse: 1.63689 | valid_rmse: 1.61696 |  0:00:06s
epoch 9  | loss: 0.00704 | train_rmse: 1.20844 | valid_rmse: 1.19085 |  0:00:07s
epoch 10 | loss: 0.00675 | train_rmse: 0.8606  | valid_rmse: 0.84984 |  0:00:08s

# Now, I repeated the same cell, without initiliazing the tabReg, rather using the already trained.
tabReg.fit(X_train=X_train, y_train=Y_train_trans,
                    eval_set=[(X_train, Y_train_trans), (X_valid, Y_valid_trans)],
                    eval_name=[‘train’, ‘valid’],
                    max_epochs=250, batch_size=256,    #bSize_opt.item(),
                    eval_metric=[‘rmse’], patience=10,  #mae, rmse
                    loss_fn = torch.nn.MSELoss(),)
And the loss history is
epoch 0  | loss: 0.22915 | train_rmse: 9.6918  | valid_rmse: 10.11569|  0:00:00s
epoch 1  | loss: 0.02308 | train_rmse: 9.36632 | valid_rmse: 9.4682  |  0:00:01s
epoch 2  | loss: 0.012   | train_rmse: 9.16359 | valid_rmse: 9.36254 |  0:00:02s
epoch 3  | loss: 0.0093  | train_rmse: 7.34215 | valid_rmse: 7.45125 |  0:00:02s
epoch 4  | loss: 0.00837 | train_rmse: 5.16882 | valid_rmse: 5.29309 |  0:00:03s
epoch 5  | loss: 0.00819 | train_rmse: 4.13819 | valid_rmse: 4.18128 |  0:00:04s
epoch 6  | loss: 0.00766 | train_rmse: 2.89976 | valid_rmse: 2.91836 |  0:00:04s
epoch 7  | loss: 0.00742 | train_rmse: 2.02085 | valid_rmse: 2.03557 |  0:00:05s
epoch 8  | loss: 0.00725 | train_rmse: 1.63689 | valid_rmse: 1.61696 |  0:00:06s
epoch 9  | loss: 0.00704 | train_rmse: 1.20844 | valid_rmse: 1.19085 |  0:00:07s
epoch 10 | loss: 0.00675 | train_rmse: 0.8606  | valid_rmse: 0.84984 |  0:00:07s

This clearly says retraining is not working.

**Comment by Optimox** [comment]
You should set  `warm_start=True` when retraining.

**Comment by HarishBaki** [comment]
@Optimox Thank you. It is working.

**Comment by Optimox** [comment]
Feel free to reopen if you have more questions related to your problem.
